app/Http/Controllers/V2/Terrafund/TerrafundEditGeometryController.php

```diff
@@ -28,6 +30,64 @@ public function getSitePolygonData(string $uuid)
         }
     }
 
+    public function updateProjectCentroid($polygonGeometry)
+    {
+        try {
+            $sitePolygon = SitePolygon::where('poly_id', $polygonGeometry->uuid)->first();
+
+            if ($sitePolygon) {
+                $relatedSite = Site::where('uuid', $sitePolygon->site_id)->first();
+                $project = Project::where('id', $relatedSite->project_id)->first();
+
+                if ($project) {
+                    $geometryHelper = new GeometryHelper();
+                    $centroid = $geometryHelper->centroidOfProject($project->uuid);
+
+                    if ($centroid === null) {
+                        Log::warning("Invalid centroid for project UUID: $project->uuid");
+                    }
+                } else {
+                    Log::warning("Project with UUID $relatedSite->project_id not found.");
+                }
+            } else {
+                Log::warning("Site polygon with UUID $polygonGeometry->uuid not found.");
+            }
```

Comment:
This could avoid a lot of nesting by using early returns. Instead of 
```
if ($thing) {
  // do stuff
} else { 
  Log::warning(...)
}
```
use
```
if (!$thing) {
  Log::warning(...);
  return null;
}

// do stuff
```

Reply:
Thanks, done for this code. I'll keep in mind for future uses.